aea/manager.py

```diff
@@ -551,20 +545,35 @@ def _make_config(
                 'Do not override any of {" ".join(self.AGENT_DO_NOT_OVERRIDE_VALUES)}'
             )
 
-        json_data = yaml_load_all(
-            AEABuilder.get_configuration_file_path(project_path).open()
+        agent_configuration_file_path: Path = AEABuilder.get_configuration_file_path(
+            project_path
         )
-        agent_config = self._update_dict(json_data[0], agent_overrides)
-
-        components_configs = {PublicId.from_json(obj): obj for obj in json_data[1:]}
-
+        loader = ConfigLoaders.from_package_type(PackageType.AGENT)
+        with agent_configuration_file_path.open() as fp:
+            agent_config: AgentConfig = loader.load(fp)
+
+        # prepare configuration overrides
+        # - agent part
+        agent_update_dictionary: Dict = dict(**agent_overrides)
+        # - components part
+        components_configs: Dict[ComponentId, Dict] = {}
         for obj in component_overrides:
-            component_id = ComponentId(obj["type"], PublicId.from_json(obj))
-            components_configs[component_id] = self._update_dict(
-                components_configs.get(component_id, {}), obj
+            obj = copy.copy(obj)
+            author, name, version = (
+                obj.pop("author"),
+                obj.pop("name"),
+                obj.pop("version"),
             )
-
-        return [agent_config] + list(components_configs.values())
+            component_id = ComponentId(obj.pop("type"), PublicId(author, name, version))
+            components_configs[component_id] = obj
+        agent_update_dictionary["component_configurations"] = components_configs
+        # do the override (and valiation)
+        agent_config.update(agent_update_dictionary)
+
+        # return the multi-paged JSON object.
+        json_data = agent_config.ordered_json
+        result: List[Dict] = [json_data] + json_data.pop("component_configurations")
```

Comment:
this was done in other places as well.

Basically, `Configuration.json` returns a dictionary; however, for `AgentConfig` the interface is not ideal. So what we do is to create a dictionary with `component_configurations` as the list of following overrides; then the burden it's up to the client of the class.